tests/c/setlongjmp.newcg.c

```diff
@@ -2,12 +2,12 @@
 // Run-time:
 //   env-var: YKD_LOG_IR=-:jit-pre-opt
 //   env-var: YKD_SERIALISE_COMPILATION=1
-//   env-var: YKD_LOG_JITSTATE=-
+//   env-var: YK_LOG=4
 //   stderr:
-//     jitstate: start-tracing
+//     yk-jit-event: start-tracing
 //     we jumped
-//     jitstate: stop-tracing
-//     jitstate: trace-compilation-aborted: longjmp encountered
+//     yk-jit-event: stop-tracing
+//     yk-warning: trace-compilation-aborted: longjmp encountered
```

Comment:
should there have been a `yk-jit-event` line informing us of a state transition (separate from the warning?)

Reply:
?

Reply:
There's a real tension with "flexibility" and "usability". I don't know what the right trade-off is: my main concern (addressed in this PR) was that we were in a real mess with warnings/errors to the extent that we didn't understand how the system was running. I've sorted that out in this PR, but I don't have further concerns myself. If someone wants to do more, they are welcome to do so, but personally I'm done for now (famous last words).